### DISCUSSION TITLE
Unassigned barcode opens supplier item

### DISCUSSION THREAD

**Discussion by dieselburner** [inventree/InvenTree#7646] [General]
I have faced rather weird issue and don't know how to fix this. I received my order from Mouser and while scanning received items one by one I have found out some of these gives error about "existing barcode". Surprisingly, scanning barcodes on these few problematic items via Android opens "supplier item", and it's the correct one. I checked the database records, and these supplier items does not have barcode fields set.
So... I assume the system somehow checks the barcode, even if it's not assigned to anything yet, and if it corresponds to supplier item, just reports with error it's already in the system.
I of course can generate my own barcode label to these received items with misbehaving barcodes, but curious if this barcode supplier check can somehow be turned off?

Just to avoid unnecessary questions: the behavior is the same if I configure Mouser barcode plugin to point to Mouser supplier, or not.

Thanks.

**Comment by dieselburner** [comment]
Clarification: I want to assign barcodes on received item bag to stock items, not supplier items.

**Comment by SchrodingersGat** [comment]
If the barcodes on the received items are not unique, then you will get this error. To assign a custom barcode to an item, it must be globally unique

**Reply by dieselburner** [reply]
That's the point, Mouser barcodes on received items are unique, as they contain:
- purcase order number which was specified during ordering 
- manufacturer 
- MPN

I will double check this, of course. But I have a gut feeling in case of duplicated barcode, scanning it would open existing stock item, which is where I assign barcodes, not supplier item, right? I never assign barcodes to supplier items or parts, just to stock.

**Reply by dieselburner** [reply]
Ok, I have found what is the problem. As per https://github.com/inventree/InvenTree/pull/5509:

> The code parses the barcode data fields and return a matching SupplierPart...

Even if the code was never scanned/entered before, there's still a chance barcode will point to some supplier part, if relevant fields match.

I tried to disable these builtin supplier barcode plugins, but apparently it is not possible. Admin interface allows to disable them, but this setting literally does nothing. Also, after docker restart the setting is not persistent and is enabled.

I guess the proper way in this case would be to generate your own stock item labels, and don't assign supplier barcodes to stock items.

**Reply by SchrodingersGat** [reply]
@dieselburner so to clarify, the issue is that you want to assign this (unique) barcode to the individual incoming stock item, but as the generic barcode fields point to an existing supplier part, this is taking priority (and preventing assignation to the stock item)?

**Reply by dieselburner** [reply]
Correct.
